apps/hash-ai-worker-ts/src/workflows/run-flow-workflow.ts

```diff
@@ -196,6 +206,81 @@ export const runFlowWorkflow = async (
 
   await flowActivities.persistFlowActivity({ flow, userAuthentication });
 
+  /**
+   * Handle requests from activities for external input (e.g. human input, HTML from a browser authenticated with a
+   * site).
+   *
+   * The process by which these requests and responses occur is:
+   * 1. Activity sends a Signal to the Workflow requesting data from the outside world when it encounters a need for it
+   * 2. The outside world is polling for these requests (resolved by finding the signals in the event history, for now)
+   * 3. The outside world sends a Signal back to the Workflow with the requested data
+   * 4. The Workflow allows the response to be accessed via a Query
+   */
```

Comment:
Nit: it might be nice to encapsulate this in a `setupHandlers` method in a separate file, to try and keep this file's size down for readability.

Reply:
Done in https://github.com/hashintel/hash/pull/4391/commits/92cf74c0598080a98b777809790d963618424d2c